### ISSUE TITLE
Uncaught TypeError: Cannot read property 'title' of undefined

### ISSUE THREAD

**Issue by stelasido** [nextapps-de/spotlight#16]
Hi, I am getting this error with the API version of using the library:

> Uncaught TypeError: Cannot read property 'title' of undefined

I am using the code from the example:

`    Spotlight.show([{
        title: "Image 1",
        description: "This is a description.",
        src: "gallery/london-1758181.jpg"
    },{
        title: "Image 2",
        description: "This is a description.",
        src: "gallery/sea-1975403.jpg"
    },{
        title: "Image 3",
        description: "This is a description.",
        src: "gallery/newport-beach-2089906.jpg"
    }]);`

* the image path doesn't matter for now.

Everything works when I use the attributes, but I need to set it up with JS.
If I console.log the Object, I see its properties: 

`autofit: ƒ bb(a)
close: ƒ eb(a)
fullscreen: ƒ Za(a)
goto: ƒ Z(a)
init: ƒ Qa()
menu: ƒ kb(a)
next: ƒ W()
play: ƒ ab(a)
prev: ƒ $a()
show: ƒ (a,b)
theme: ƒ Na(a)
zoom: ƒ nb(a)`

**Comment by joandavi** [comment]
did you manage to solve the problem? i got the same

**Comment by chris-makaio** [comment]
maybe y'all got it solved, but it just happened to me..
Alright, it depends on how and when you inject script. Spotlight adds a div with class "pane" on "DOMContentLoaded" which is referenced and undefined in the function which raises the exception. Obviously you'll have to take care of the script injection yourselves.

I fixed it in the (current master) bundled version like this:

Search for 
```javascript
P&&V(!1);K.parentNode.removeChild(K);
```

and change it to 

```javascript
P&&V(!1);K.parentNode&&K.parentNode.removeChild(K);
```

then, when you're ready to start the script, just call Spotlight.init like this
```javascript
    let spotlightScript = document.createElement('script');
    spotlightScript.src = "/scripts/spotlight.bundle.js";
    spotlightScript.async = true;
    spotlightScript.onload = () => Spotlight.init();
```

**Comment by ryzexsp** [comment]
same , chris answer didnt help

**Comment by web56** [comment]
> did you manage to solve the problem? i got the same

Is there any solution as I'm faing the same error! PLEASE Help

**Comment by web56** [comment]
> Hi, I am getting this error with the API version of using the library:
> 
> > Uncaught TypeError: Cannot read property 'title' of undefined
> 
> I am using the code from the example:
> 
> ` Spotlight.show([{ title: "Image 1", description: "This is a description.", src: "gallery/london-1758181.jpg" },{ title: "Image 2", description: "This is a description.", src: "gallery/sea-1975403.jpg" },{ title: "Image 3", description: "This is a description.", src: "gallery/newport-beach-2089906.jpg" }]);`
> 
> * the image path doesn't matter for now.
> 
> Everything works when I use the attributes, but I need to set it up with JS.
> If I console.log the Object, I see its properties:
> 
> `autofit: ƒ bb(a) close: ƒ eb(a) fullscreen: ƒ Za(a) goto: ƒ Z(a) init: ƒ Qa() menu: ƒ kb(a) next: ƒ W() play: ƒ ab(a) prev: ƒ $a() show: ƒ (a,b) theme: ƒ Na(a) zoom: ƒ nb(a)`

Is there any solution as I'm haing the same error! PLEASE Help

**Comment by chris-makaio** [comment]
@web56 
No, I switched to another lib

**Comment by ts-thomas** [comment]
The programmatic usage was slightly broken. The focus was basically on the markup-only paradigm. That all is now fixed/improved in the new version >= 0.7.0.
